index.js

```diff
@@ -49,7 +70,13 @@ export default class AWSHttpProvider extends HttpProvider {
     try {
       const strPayload = JSON.stringify(payload);
       const region = process.env.AWS_DEFAULT_REGION || 'us-east-1';
-      let credentials = new AWS.EnvironmentCredentials('AWS');
+      let credentials
+      if (!!this.ssmCredentials) {
+        if (!validateCredentials(this.ssmCredentials)) throw 'Invalid Credentials: Check that your AWS credentials match the standard format';
+        credentials = new AWS.Credentials(this.ssmCredentials);
+      } else {
+        credentials = new AWS.EnvironmentCredentials('AWS');
+      }
```

Comment:
Recommend instead:
```
const creds =
  'ssmCredentials' in this &&
  'accessKeyId' in this.ssmCredentials &&
  'secretAccessKey' in this.ssmCredentials &&
  this.ssmCredentials;
const credentials = (creds && new AWS.Credentials(creds)) || new AWS.EnvironmentCredentials('AWS');
```

Reply:
required minor change to constructor as well to set to a null object instead of a boolean, but works well. Removing method `validateCredentials()`